### PR TITLE
DAPI-1392: describe quality score options in API doc

### PR DESCRIPTION
**Pull request by rouliane**

### CODE REVIEW

**Review by AymericPlanche — APPROVED** [review]

File: content/swagger/resources/products/definitions/product.yaml [hunk 1]
```diff
@@ -134,6 +134,9 @@ properties:
         description: Status of the product regarding the user permissions (only available since the v2.0 in the Enterprise Edition)
         type: string
         enum: ['read_only','draft_in_progress','proposal_waiting_for_approval','working_copy']
+  quality_scores:
+    type: object
+    description: Only available when query parameter "with_quality_scores" is set to "true"
```
Comment (fitn) [inline content/swagger/resources/products/definitions/product.yaml:139]:
```suggestion
    description: Only available when query parameter "with_quality_scores" is set to "true" (only available in Serenity)
```

**Review by fitn — APPROVED** [review]
1 comment in order to better define from which version it will be available :)

File: content/swagger/resources/products/definitions/product.yaml (hunk 1 above)
Comment (NolwennP) [reply content/swagger/resources/products/definitions/product.yaml:139]: And maybe, actually, describe what the object will contain. Something like this:
```suggestion
    description: Product quality scores for each channel/locale combination (only available in Serenity and when the "with_quality_scores" query parameter is set to "true")
```

File: content/swagger/parameters.yaml [hunk 2]
```diff
@@ -92,3 +92,10 @@ with_attribute_options:
   default: false
   type: boolean
   required: false
+with_quality_scores:
+  name: with_quality_scores
+  in: query
+  description: Return product's quality scores in the response.
```
Comment (NolwennP) [inline content/swagger/parameters.yaml:98]:
```suggestion
  description: Return product quality scores in the response.
```

File: content/swagger/resources/products/definitions/product.yaml [hunk 3]
```diff
@@ -134,6 +134,9 @@ properties:
...
```
Comment (NolwennP) [inline content/swagger/resources/products/definitions/product.yaml:140]:
```suggestion
    x-immutable: true
    x-read-only: true

```

File: content/swagger/resources/products/definitions/product.yaml [hunk 4]
```diff
@@ -134,6 +134,11 @@ properties:
         description: Status of the product regarding the user permissions (only available since the v2.0 in the Enterprise Edition)
         type: string
         enum: ['read_only','draft_in_progress','proposal_waiting_for_approval','working_copy']
+  quality_scores:
+    type: object
+    description: Only available when query parameter "with_quality_scores" is set to "true" (only available in Serenity)
```
Comment (NolwennP) [inline content/swagger/resources/products/definitions/product.yaml:139]: As this is a description, you need to actually describe what the object will contain.
```suggestion
    description: Product quality scores for each channel/locale combination (only available in Serenity and when the "with_quality_scores" query parameter is set to "true")
```

**Review by NolwennP — COMMENTED** [review]
Two things:
- In the Swagger, an actual description should be given in the `description` field for this new property. See my comment.
- As you enrich the JSON format of a product, you also need to update it here: https://api.akeneo.com/concepts/products.html#product Add the new property in the JSON example, and add a warning box below stating that this field is only available in Serenity and when the "with_quality_scores" query parameter is set to true.

File: content/concepts/products.md [hunk 5]
```diff
@@ -111,6 +118,10 @@ Note that the `metadata` field is only available starting from the v2 and as it
 Note that the `quantified_associations` field is only available in Serenity.
 :::
 
+::: warning
+Note that the `quality_scores` field is only available in Serenity and when the "with_quality_scores" query parameter is set to "true".
```
Comment (NolwennP) [inline content/concepts/products.md:122]:
```suggestion
Note that the `quality_scores` field is only available in Serenity and when the `with_quality_scores` query parameter is set to `true`.
```

**Review by NolwennP — APPROVED** [review]

**Pull request merged**